server/src/main/java/com/thoughtworks/go/server/service/ElasticAgentPluginService.java

```diff
@@ -210,7 +213,8 @@ public String getPluginStatusReport(String pluginId) {
     public String getAgentStatusReport(String pluginId, JobIdentifier jobIdentifier, String elasticAgentId) {
         final ElasticAgentPluginInfo pluginInfo = elasticAgentMetadataStore.getPluginInfo(pluginId);
         if (pluginInfo.getCapabilities().supportsAgentStatusReport()) {
-            return elasticAgentPluginRegistry.getAgentStatusReport(pluginId, jobIdentifier, elasticAgentId);
+            ClusterProfile clusterProfile = jobInstanceSqlMapDao.loadPlan(jobIdentifier.getId()).getClusterProfile();
```

Comment:
There is a possibility of `ClusterProfile` being null as the job plan gets deleted as soon as the job finishes.
Calling `clusterProfile.getConfigurationAsMap(true)` on `null` might result in NullPointerException.

Please verify :)